### PR TITLE
brief clean up of aero_model_init and mam_prevap_resusp_init

### PR DESCRIPTION
**Pull request by tangshuaiqi**
only made simple clean up to remove unused variables and namelist options. Not sure how C++ team will port these `_init` subroutines but I can always further refactor them if needed.

### CODE REVIEW

**Review by jeff-cohere — APPROVED** [review]

**Review by mingxuanwupnnl — APPROVED** [review]
The code changes look good to me. history_verbose is an important namelist flag, it is often turned on when doing our own runs. Of course, we can just remove it and add them back later. I'm confused from time to time how we should deal with that. Note that, these output are not validated BFB, since it is always false.
In my refactored nucleate_ice_cam.F90, they already add some diagnostic outputs (e.g., NIHF) to C++ which are not  turned on by default. @singhbalwinder do we need some consensus on this?

File: components/eam/src/chemistry/modal_aero/aero_model.F90 [hunk 1]
```diff
@@ -202,6 +198,20 @@ subroutine aero_model_init( pbuf2d, species_class, iflagaa )
...
+    !    (2) convproc_do_aer=.false. and convproc_do_gas=.true. is no longer allowed.
+    !
+    ! for C++ porting: namelist [history_aerosol, history_verbose, convproc_do_aer, resus_fix]
+    ! are [T, F, T, T] in default. history_aerosol is kept as there is no code
+    ! related to history_aerosol=.false. All other conditions are removed
+    ! as they are not used/tested.
+    if (history_verbose) call endrun('history_verbose=.true. is removed')
+    if (.not. convproc_do_aer) call endrun('convproc_do_aer=.false. is removed')
+    if (.not. resus_fix) call endrun('resus_fix=.false. is removed')
+
```
Comment (singhbalwinder) [inline components/eam/src/chemistry/modal_aero/aero_model.F90:213]: We might want to keep the history namelist variables. I was going to add a test to turn these on just to make sure we are not messing up any diagnostics.

**Review by singhbalwinder — CHANGES_REQUESTED** [review]
We can discuss it more but I think we should keep the history related flags.

File: components/eam/src/chemistry/modal_aero/aero_model.F90 (hunk 1 above)
Comment (tangshuaiqi) [reply components/eam/src/chemistry/modal_aero/aero_model.F90:213]: did you mean to keep only `history_verbose` or all the three namelist variables?

Comment (singhbalwinder) [reply components/eam/src/chemistry/modal_aero/aero_model.F90:213]: Just keep the `history_verbose`.

**Comment by tangshuaiqi** [comment]
@singhbalwinder the history_verbose flag is added back. please review the latest change.

**Review by singhbalwinder — APPROVED** [review]
Looks good!

**Review by brigaudet — APPROVED** [review]

**Comment by tangshuaiqi** [comment]
All tests passed. merge into maint

comp_testall_yaml_aero_model_init_mam4_org_v2_baselines: 9 tests
  PASS ERP_P64x1.ne4pg2_oQU480.F2010.compy_intel
  PASS ERS.ne4pg2_oQU480.F2010.compy_intel
  PASS PEM_P64x1.ne4pg2_oQU480.F2010.compy_intel
  PASS PET.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS_D_Ln5_P1x1.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS_D_Ln5_P32x1.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS_D.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS_Ly1.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS.ne30pg2_EC30to60E2r2.F2010.compy_intel

**Pull request merged**